elements/dash-accordion/dash-accordion.js

```diff
@@ -0,0 +1,529 @@
+      children.forEach(child => {
+        child.style.position = 'relative';
+        // All children _before_ the animated ones need to be painted _over_
+        // all the animated children. Therefore, all the non-animated children
+        // get `z-index: 2`;
+        child.style.zIndex = 2;
+      });
+
+      // All the animated children get `z-index: 0` and get translated to the
+      // start position. Because this is a CSS transition we don’t need to
+      // use `will-change`.
+      animatedChildren.forEach(child => {
+        child.style.position = 'relative';
+        child.style.zIndex = 1;
+        child.style.transform = `translateY(${startOffset}px)`;
+      });
+      // Wait a frame to apply the styles.
+      return requestAnimationFramePromise()
+        .then(_ => {
+          // Set all animated children to their end position.
+          animatedChildren.forEach(child => {
+            child.style.transform = `translateY(${endOffset}px)`;
+            child.classList.add('animating');
+          });
+        })
+        // Wait for the transition to finish
+        .then(_ => transitionEndPromise(panel))
+        .then(_ => {
+          // Clean up all the temporary styles
+          animatedChildren.forEach(child => {
+            child.style.transform = '';
+            child.classList.remove('animating');
+          });
+          children.forEach(child => {
+            child.style.position = '';
+            child.style.zIndex = '';
+          });
+          this.style.overflow = '';
+        });
+    }
+  }
+  window.customElements.define('dash-accordion', DashAccordion);
+
+  // `dashAccordionHeadingCounter` counts the number of
+  // `<dash-accordion-heading>` instances created. The
+  // number is used to generated new, unique IDs.
+  let dashAccordionHeadingCounter = 0;
+  /**
+   * `DashAccordionHeading` is a heading for a `<dash-accordion>`.
+   * `<dash-accordion-heading>` should always be used with `role=heading` in the
+   * markup so that the semantics remain useable when JavaScript is failing.
+   *
+   * A `<dash-accordion-heading>` declares which `<dash-accordion-panel>` it
+   * belongs to by using that panel’s ID as the value for the `aria-controls`
+   * attribute.
+   *
+   * A `<dash-accordion-heading>` will automatically generate a unique ID if
+   * none is specified.
+   */
+  class DashAccordionHeading extends HTMLElement {
+    constructor() {
+      super();
+    }
+
+    connectedCallback() {
+      // In contrast to `<dash-tabs>`, the children handle the input events.
+      this.addEventListener('keydown', this._onKeyDown);
+      this.addEventListener('click', this._onClick);
+
+      this.setAttribute('role', 'heading');
+      // The WAI-ARIA best practices prescribe a button inside the heading. To
+      // keep the markup simple, `<dash-accordion-heading>` behaves like a
+      // button instead by being focusable and reacting to enter and space.
```

Comment:
So I'm not a bottleneck on this I'd suggest you join the web a11y slack and ask in #general

http://web-a11y.herokuapp.com/